File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if root:
            output = [root.val]
        else: return []
        
        if root.left:
            output = output + self.preorderTraversal(root.left)
        if root.right:
            output = output + self.preorderTraversal(root.right)
        
        return output

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func preorderTraversal(root *TreeNode) []int {
    var dfs func(root *TreeNode)
    result := []int{}
    dfs = func(root *TreeNode) {
        if root == nil {
            return
        }
        result = append(result, root.Val)
        dfs(root.Left)
        dfs(root.Right)
    }
    dfs(root)
    return result
}